### PR TITLE
Test automation

### PR DESCRIPTION
**Pull request by Seequick1**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

System test, which checking if editing labels via Kafka CR will be derived to services, config maps, stateful set and also Kafka pod itself.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Seequick1** [comment]
@strimzi-ci run tests testcase=KafkaST#testLabelModificationDoesNotBreakCluster

File: systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ void testRecoveryDuringZookeeperRollingUpdate() {
         String reconciliation = TimeMeasuringSystem.startOperation(Operation.NEXT_RECONCILIATION);
 
         LOGGER.info("Wait till another rolling update starts");
-        TestUtils.waitFor("Wait till another rolling update starts", Constants.CO_OPERATION_TIMEOUT_POLL, Constants.CO_OPERATION_TIMEOUT,
+        TestUtils.waitFor("Wmy-cluster-kafka-0ait till another rolling update starts", Constants.CO_OPERATION_TIMEOUT_POLL, Constants.CO_OPERATION_TIMEOUT,
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/RollingUpdateST.java]: Typo?

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java [hunk 2]
```diff
@@ -1460,6 +1466,195 @@ void checkStorageSizeForVolumes(List<PersistentVolumeClaim> volumes, String[] di
...
+        StatefulSet statefulSet = kubeClient().getStatefulSet(kafkaClusterName(CLUSTER_NAME));
+        labels = statefulSet.getSpec().getTemplate().getMetadata().getLabels();
+        LOGGER.info("Verifying default labels in the KafkaCR");
+
+        assertEquals(labelValues[0], labels.get(labelKeys[0]));
+        assertEquals(labelValues[1], labels.get(labelKeys[1]));
+
+        LOGGER.info("Edit kafka labels in Kafka CR");
+        replaceKafkaResource(CLUSTER_NAME, resource -> {
+            resource.getMetadata().getLabels().put(labelKeys[0], labelNewValues[0]);
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java]: This seems to only change the existing labels. Can we also:
* Add a new label
* Remove one of the old labels?

Comment (Seequick1) [reply systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java]: Of course we can. Thanks for the suggestion. Done.

**Comment by Seequick1** [comment]
@strimzi-ci run tests testcase=KafkaST#testLabelModificationDoesNotBreakCluster

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java [hunk 3]
```diff
@@ -1547,128 +1551,142 @@ void checkStorageSizeForVolumes(List<PersistentVolumeClaim> volumes, String[] di
         }
     }
 
-    @Tag(SCALABILITY)
     @Test
-    void testBigAmountOfTopicsCreatingViaK8s() {
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java]: Are you sure this doesn't delete these tests now?

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java [hunk 4]
```diff
@@ -1547,128 +1551,142 @@ void checkStorageSizeForVolumes(List<PersistentVolumeClaim> volumes, String[] di
...
-            StUtils.waitForKafkaTopicPartitionChange(currentTopic, topicPartitions);
-            verifyTopicViaKafka(currentTopic, topicPartitions);
-        }
-    }
+        assertEquals(labelValues[0], labels.get(labelKeys[0]));
+        assertEquals(labelValues[1], labels.get(labelKeys[1]));
 
-    @Tag(SCALABILITY)
-    @Test
-    void testBigAmountOfTopicsCreatingViaKafka() {
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java]: Are you sure this doesn't delete these tests now?

File: systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java [hunk 5]
```diff
@@ -1547,128 +1551,142 @@ void checkStorageSizeForVolumes(List<PersistentVolumeClaim> volumes, String[] di
...
-        assertNotNull(kafkaTopic);
-        assertThat(listTopicsUsingPodCLI(CLUSTER_NAME, 0), hasItem(topicName));
-        assertEquals(topicName, kafkaTopic.getMetadata().getName());
-        assertEquals(topicPartitions, kafkaTopic.getSpec().getPartitions());
-    }
+        LOGGER.info("Waiting for kafka config map labels changed {}", labels);
+        StUtils.waitForKafkaConfigMapLabelsChange(configMapName, labels);
 
-    @Test
-    void testRegenerateCertExternalAddressChange() throws InterruptedException {
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java]: Are you sure this doesn't delete these tests now?

**Comment by scholzj** [comment]
Replaced by #1863.

**Pull request closed without merging**